src/search/odoc_html_frontend.ml

```diff
@@ -19,9 +104,10 @@ let of_strings ~kind ~prefix_name ~name ~rhs ~typedecl_params ~doc =
         ]
   and prefix_name =
     match prefix_name with
+    | None -> []
+    | Some "" -> []
     | Some prefix_name ->
         [ span ~a:[ a_class [ "prefix-name" ] ] [ txt (prefix_name ^ ".") ] ]
```

Comment:
There is indeed a bug here. But wouldn't it be better to remove the addition of `"."` to the prefix name:

```suggestion
  and prefix_name =
    match prefix_name with
    | None -> []
    | Some prefix_name ->
        [ span ~a:[ a_class [ "prefix-name" ] ] [ txt prefix_name ] ]
```
and modify [`names_of_id`](https://github.com/ocaml/odoc/blob/master/src/search/html.ml#L174) accordingly?

Reply:
Yes, it is better i agree.

Reply:
On second thought, no its not better. I believe there a few place where we String.split the prefix name. This would behave very poorly

Reply:
In sherlodoc or in odoc? Could you provide links to such places?

That would maybe mean that the prefix name is turned into a string too early, and we should keep it as a list?

Reply:
We should probably keep it as a list.

Reply:
I went and checked more closely. We do indeed use String.split on the name, but we do not use the name functions provided by odoc.search, we call fullname which is a list, so it does not change anything to switch that around. We could even say that the bug was sherlodoc's fault, we could have given None instead of Some "".